resolwe/flow/models/fields.py

```diff
@@ -0,0 +1,91 @@
+"""Custom database fields."""
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+from six import string_types
+
+from django.db import connection
+from django.db.models.fields import SlugField
+from django.utils.text import slugify
+
+
+class ResolweSlugField(SlugField):
+    """Slug field."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize field."""
+        self.populate_from = kwargs.pop('populate_from', None)
+
+        # unique_with value can be string or tuple
+        self.unique_with = kwargs.pop('unique_with', ())
+        if isinstance(self.unique_with, string_types):
+            self.unique_with = (self.unique_with,)
+
+        if self.unique_with:
+            # we will do "manual" granular check below
+            kwargs['unique'] = False
+
+        self.always_update = kwargs.pop('always_update', False)
+
+        super(ResolweSlugField, self).__init__(*args, **kwargs)
+
+    def pre_save(self, instance, add):
+        """Ensure slug uniqunes before save."""
+        model = self.model
+        slug = self.value_from_object(instance)
+
+        if self.populate_from and not slug:
+            if hasattr(self.populate_from, '__call__'):
+                # ResolweSlugField(populate_from=lambda instance: ...)
+                slug = self.populate_from(instance)
+            else:
+                # ResolweSlugField(populate_from='foo')
+                attr = getattr(instance, self.populate_from)
+                slug = attr() if callable(attr) else attr
+
+        if slug:
+            slug = slugify(slug)
+
+        if not slug:
+            slug = None
+
+            if not self.blank:
+                slug = instance._meta.model_name  # pylint: disable=protected-access
+            elif not self.null:
+                slug = ''
+
+        if slug:
+            # Make sure that potential index added won't excede maximal length.
+            slug = slug[:(self.max_length - 10)]
+
+            query_params = {
+                'slug': slug,
+                'slug_column': self.column,
+                'table_name': model._meta.db_table,  # pylint: disable=protected-access
+            }
+
+            with connection.cursor() as cursor:
+                cursor.execute(
+                    """
+                    SELECT MAX(slug_index) + 1 FROM (
+                        SELECT COALESCE(
+                            NULLIF(
+                                RIGHT({slug_column}, -length('{slug}')-1),
+                                ''
+                            ),
+                            '0'
+                        )::text::integer AS slug_index
+                        FROM {table_name} WHERE (
+                            {slug_column} ~ '^{slug}(-[0-9]*)?$'
+                        )
+                    ) AS tmp
+                    """.format(**query_params)
```

Comment:
Yikes! Never do this as you just introduced an SQL injection vulnerability. Also, why don't you use Django's raw query instead of accessing the underlying backend (psycopg2) directly?